iwyu_path_util.cc

```diff
@@ -149,6 +157,9 @@ string NormalizeFilePath(const string& path) {
   string result = CanonicalizeFilePath(path);
   while (StripLeft(&result, "./")) {
   }
+  // Remove trailing './' but don't split '../'
```

Comment:
I wonder if we should now remove `NormalizeFilePath` entirely, and move this stuff to `CanonicalizeFilePath` instead?

`llvm::sys::path` has a `remove_dots` method that collapses all dot components, as far as I can tell:
http://llvm.org/docs/doxygen/html/namespacellvm_1_1sys_1_1path.html#a07079865968cc0fac090076bd8681750

If we move that into `CanonicalizeFilePath`, life should be better overall, I think?